### PR TITLE
implement two API endpoints, user_login and user_retrieve

### PR DESCRIPTION
**Pull request by SoniaChoo**
implement two user-related API, login in and retrieve user by user_id

### CODE REVIEW

File: db/user.go [hunk 1]
```diff
@@ -27,79 +29,51 @@ func InsertUser(u model.User) error {
 	return nil
 }
 
-// TODO: implement update function.
-func UpdateUser(user, newUser model.User) error {
+func LoginUser(u *model.User) error {
 	db, err := DBFactory()
 	if err != nil {
-
+		log.Printf("error connect database, %v\n", err)
```
Comment (ziyitony) [inline db/user.go:35]: when an error happens, please remember to `return err` to stop the following process.

File: http/userinfoHandler.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ import (
 
 // userinfoHandler is the function for inputing userinfo,
 // parameters: name,age, gender, address.
-func UserinfoHandler(w http.ResponseWriter, r *http.Request) {//zhuzhu:更新了函数名
+func UserinfoHandler(w http.ResponseWriter, r *http.Request) { //zhuzhu:更新了函数名
```
Comment (ziyitony) [inline http/userinfoHandler.go]: please delete this file.

**Review by ziyitony — CHANGES_REQUESTED** [review]
please address the comments.

File: db/user.go (hunk 1 above)
Comment (SoniaChoo) [reply db/user.go:35]: updated.

File: http/userinfoHandler.go (hunk 2 above)
Comment (SoniaChoo) [reply http/userinfoHandler.go]: updated.

**Review by ziyitony — APPROVED** [review]

**Pull request merged**